internal/cmd/skupper/listener/kube/listener_status.go

```diff
@@ -0,0 +1,156 @@
+	var validationErrors []error
+	resourceStringValidator := validator.NewResourceStringValidator()
+	outputTypeValidator := validator.NewOptionValidator(utils.OutputTypes)
+
+	// Validate arguments name if specified
+	if len(args) > 1 {
+		validationErrors = append(validationErrors, fmt.Errorf("only one argument is allowed for this command"))
+	} else if len(args) == 1 {
+		if args[0] == "" {
+			validationErrors = append(validationErrors, fmt.Errorf("listener name must not be empty"))
+		} else {
+			ok, err := resourceStringValidator.Evaluate(args[0])
+			if !ok {
+				validationErrors = append(validationErrors, fmt.Errorf("listener name is not valid: %s", err))
+			} else {
+				cmd.name = args[0]
+			}
+		}
+	}
+
+	// Validate that there is no listener with this name in the namespace
+	if cmd.name != "" {
+		listener, err := cmd.client.Listeners(cmd.namespace).Get(context.TODO(), cmd.name, metav1.GetOptions{})
+		if listener == nil || errors.IsNotFound(err) {
+			validationErrors = append(validationErrors, fmt.Errorf("listener %s does not exist in namespace %s", cmd.name, cmd.namespace))
+		}
+	}
+
+	if cmd.flags.output != "" {
+		ok, err := outputTypeValidator.Evaluate(cmd.flags.output)
+		if !ok {
+			validationErrors = append(validationErrors, fmt.Errorf("output type is not valid: %s", err))
+		} else {
+			cmd.output = cmd.flags.output
+		}
+	}
+
+	return validationErrors
+}
+func (cmd *CmdListenerStatus) Run() error {
+	if cmd.name == "" {
+		resources, err := cmd.client.Listeners(cmd.namespace).List(context.TODO(), metav1.ListOptions{})
+		if err != nil || resources == nil || len(resources.Items) == 0 {
+			fmt.Println("No listeners found")
+			return err
+		}
+		if cmd.output != "" {
+			for _, resource := range resources.Items {
+				encodedOutput, err := utils.Encode(cmd.output, resource)
+				if err != nil {
+					return err
+				}
+				fmt.Println(encodedOutput)
+			}
+		} else {
+			tw := tabwriter.NewWriter(os.Stdout, 8, 8, 1, '\t', tabwriter.TabIndent)
+			_, _ = fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%s\t%s\t%s\t%s",
+				"NAME", "STATUS", "ROUTING-KEY", "HOST", "PORT", "MATCHING-CONNECTORS"))
+			for _, resource := range resources.Items {
+				fmt.Fprintln(tw, fmt.Sprintf("%s\t%s\t%s\t%s\t%d",
+					resource.Name, resource.Status.StatusMessage, resource.Spec.RoutingKey, resource.Spec.Host, resource.Spec.Port))
+			}
+			_ = tw.Flush()
+		}
+	} else {
+		resource, err := cmd.client.Listeners(cmd.namespace).Get(context.TODO(), cmd.name, metav1.GetOptions{})
+		if resource == nil || errors.IsNotFound(err) {
+			fmt.Println("No listeners found")
+			return err
+		}
+		if cmd.output != "" {
+			encodedOutput, err := utils.Encode(cmd.output, resource)
+			if err != nil {
+				return err
+			}
+			fmt.Println(encodedOutput)
+		} else {
+			tw := tabwriter.NewWriter(os.Stdout, 8, 8, 1, '\t', tabwriter.TabIndent)
+			fmt.Fprintln(tw, fmt.Sprintf("Name:\t%s\nStatus:\t%s\nRouting key:\t%s\nHost:\t%s\nPort:\t%d\nConnectors:\t1\n",
```

Comment:
This value should be removed

Reply:
yes that was just a place holder.    I wasn't sure how to figure out how many connectors